### PR TITLE
Service implementations

### PR DESCRIPTION
**Pull request by Anlord241**
Created implementations for some services (every service has an interface + its implementation now)

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/config/WebSecurityConfiguration.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.team4.testingsystem.config;
 
 import com.team4.testingsystem.security.JwtRequestFilter;
-import com.team4.testingsystem.services.CustomUserDetailsService;
+import com.team4.testingsystem.services.impl.CustomUserDetailsService;
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/config/WebSecurityConfiguration.java]: All the classes in services.impl package have Impl postfix. Let's use it here as well

File: src/main/java/com/team4/testingsystem/controllers/TestsController.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 public class TestsController {
 
     @Autowired
-    private TestsService testsService;
+    private TestsServiceImpl testsService;
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/TestsController.java]: Let's use constructor injection here

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]
please, fix the comments

**Review by AntonMaksimenkau — APPROVED** [review]

**Review by maxim12321 — APPROVED** [review]

File: src/main/java/com/team4/testingsystem/controllers/TestsController.java [hunk 3]
```diff
@@ -17,12 +17,11 @@
 @RestController
 @RequestMapping(path = "/tests")
 public class TestsController {
-
-
-    private TestsService testsService;
+    
+    private TestsServiceImpl testsService;
```
Comment (maxim12321) [inline src/main/java/com/team4/testingsystem/controllers/TestsController.java]: Why `Impl` here?

File: src/main/java/com/team4/testingsystem/controllers/TestsController.java [hunk 4]
```diff
@@ -17,12 +17,11 @@
...
 
     @Autowired
-    public TestsController(TestsService testsService) {
+    public TestsController(TestsServiceImpl testsService) {
```
Comment (maxim12321) [inline src/main/java/com/team4/testingsystem/controllers/TestsController.java]: And here

**Review by maxim12321 — CHANGES_REQUESTED** [review]
Why all dependencies now have `Impl` type? I think we should autowire interfaces, not implementations...

File: src/main/java/com/team4/testingsystem/services/impl/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private final AuthenticationManager authenticationManager;
+
+    private final CustomUserDetailsServiceImpl userDetailsService;
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/AuthenticationServiceImpl.java]: still impl here

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

**Review by maxim12321 — CHANGES_REQUESTED** [review]
I think that in test files should be only one `Impl` - the implementation being tested. All other `@Mock` objects also should be interfaces.

**Review by maxim12321 — APPROVED** [review]

**Pull request merged**